### PR TITLE
fix: Fix signing requirements for updating metadata on Token

### LINKED ISSUE #12387: Metadata HIPs : Fix the signing requirements for updating `metadata` on `TokenUpdate`
Currently the signing requirements for updating `metadata` field on `TokenUpdate` only require `adminKey` of the token to sign the transaction.
But the signing requirements should require either `adminKey` **or** `metadataKey` signature

So a TokenUpdate to metadata field should succeed if there is `adminKey` **or** `metadataKey` signature. If none is present should fail the transaction

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
Fixes #12387 

- Fixes signing requirements for TokenUpdate to require either adminKey or metadataKey, when updating `metadata` field
- Also fixes signing requirements for expiry only update. Previously it failed with `INVALID_SIGNATURE` if there is no adminKey signature for expiry only update operation

### CODE REVIEW

**Review by iwsimon — DISMISSED** [review]
LGTM. Thanks @Neeharika-Sompalli

**Review by tinker-michaelj — APPROVED** [review]
LGTM, tyvm @Neeharika-Sompalli !

**Review by mhess-swl — APPROVED** [review]

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenUpdateHandler.java [hunk 1]
```diff
@@ -395,9 +418,13 @@ private void updateKeys(
             validateTrue(originalToken.hasMetadataKey(), TOKEN_HAS_NO_METADATA_KEY);
             builder.metadataKey(op.metadataKey());
         }
-        if (!isExpiryOnlyUpdateOp(op)) {
-            validateTrue(originalToken.hasAdminKey(), TOKEN_IS_IMMUTABLE);
+
+        if (isMetadataOnlyUpdateOp(op)) {
+            validateTrue(originalToken.hasAdminKey() || originalToken.hasMetadataKey(), TOKEN_IS_IMMUTABLE);
+        } else if (!isExpiryOnlyUpdateOp(op)) {
+            validateFalse(originalToken.hasAdminKey(), TOKEN_IS_IMMUTABLE);
```
Comment (netopyr) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenUpdateHandler.java:425]: Shouldn't this be `validateFalse(isEmpty(token.adminKey()), TOKEN_IS_IMMUTABLE);`?

Are these checks necessary at all? We have already done them in `validateSemantics()`.

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenUpdateHandler.java [hunk 2]
```diff
@@ -116,6 +119,27 @@ public void preHandle(@NonNull final PreHandleContext context) throws PreCheckEx
         if (op.hasAdminKey()) {
             context.requireKey(op.adminKeyOrThrow());
         }
+        // To update metadata either admin key or metadata key should sign.
```
Comment (netopyr) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenUpdateHandler.java:122]: This is unrelated to this PR, but shouldn't we require the signatures of all new keys if they are changed?

Comment (Neeharika-Sompalli) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenUpdateHandler.java:122]: No any key changes **except** adminKey require only adminKey signature. But we have checks that keys are not empty and already exist on token.

**Review by tinker-michaelj — APPROVED** [review]
LGTM, ty @Neeharika-Sompalli !

**Review by tinker-michaelj — APPROVED** [review]

**Review by thomas-swirlds-labs — APPROVED** [review]
LGTM! Thanks @Neeharika-Sompalli

**Review by iwsimon — APPROVED** [review]
LGTM. Thanks @Neeharika-Sompalli

**Pull request merged**